clusterman/autoscaler/pool_manager.py

```diff
@@ -80,6 +81,10 @@ def __init__(
             self.pool_config.read_int("scaling_limits.min_node_scalein_uptime_seconds", default=-1),
             MAX_MIN_NODE_SCALEIN_UPTIME_SECONDS,
         )
+        autoscaling_config = get_autoscaling_config(
+            POOL_NAMESPACE.format(pool=self.pool, scheduler=self.scheduler),
+        )
+        self.orphan_instance_uptime_threshold_seconds = autoscaling_config.orphan_instance_uptime_threshold_seconds
```

Comment:
I don't know if re-using this parameter is a good idea, as this indicates the maximum time after which an orphan node should be considered unrepairable, while you are looking for a minimum grace period. Don't know if we may end up having any problem with it.

Also, we should probably rename `min_node_scalein_uptime_seconds ` to something like `min_node_scalein_uptime_priority_seconds`, since that goes to inform selection sorting rather than filtering (my bad for not being more explicit when I introduced it)

Reply:
Actually I think again that we don't need this setting. We can mark orphan non-killable since "terminate_expired_orphan_instances" will terminate them. What do you think?

Reply:
But then if a pool has `recycle_orphan_instances` set to False those node would end up living forever, wouldn't they?

Reply:
Yeah, perfect catch!

Reply:
> don't know if re-using this parameter is a good idea, as this indicates the maximum time after which an orphan node should be considered unrepairable, while you are looking for a minimum grace period. Don't know if we may end up having any problem with it.

Actually they are similar (maybe same). We just need threshold to kill orphan after it reached.

Reply:
I thought again about orphan instances. We shouldn’t mark them as killable. Our cleaning logic will clean orphans. Yeap some pool owners can disable it, but there is no reason for it (who loves to keep orphan instances in pool?). I implemented flag for smooth deployment. Even I can remove flag. Additionally we have alert for orphan instances in case of necessity.